### PR TITLE
Increase quicktest code coverage by adding the following unused XenAPI calls

### PR DESCRIPTION
**Pull request by minishrink**
Increase quicktest code coverage by adding the following unused XenAPI calls:

- Client.PBD.get_all
- Client.PBD.get_uuid
- Client.PBD.get_by_uuid

Signed-off-by: Akanksha Mathur <akanksha.mathur@citrix.com>

### CODE REVIEW

File: ocaml/xapi/quicktest_bvt.ml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  let with_pbd fn pbd = with_api fn ~self:pbd in
+
+  (* checks get_uuid returns correct UUID by comparing
+   * the PBD returned by get_by_uuid *)
+  let uuid_test pbd =
+    let uuid_t = make_test "Testing 'get_by_uuid', 'get_uuid'" 4 in
+    try
+      start uuid_t;
+      let uuid = with_pbd P.get_uuid pbd in
+      if with_api P.get_by_uuid ~uuid = pbd then success uuid_t
```
Comment (lindig) [inline ocaml/xapi/quicktest_bvt.ml]: The indentation of `if..then..else` is not following the logical structure. Please use a more conventional way.

File: ocaml/xapi/quicktest_bvt.ml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    try
+      start test;
+      let pbd = List.hd (with_api P.get_all) in
+      uuid_test pbd;
+      debug test "PBD verification complete";
+      success test;
+    with
+    | (Failure hd) -> failed test "Error: no PBD available"
+    | e -> debug test (ExnHelper.string_of_exn e);
+           failed test ""
```
Comment (lindig) [inline ocaml/xapi/quicktest_bvt.ml]: This should provide useful information. I would suggest:

```
| e -> failed test (ExnHelper.string_of_exn e)
```

File: ocaml/xapi/quicktest_bvt.ml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  (* checks get_uuid returns correct UUID by comparing
+   * the PBD returned by get_by_uuid *)
+  let uuid_test pbd =
+    let uuid_t = make_test "Testing 'get_by_uuid', 'get_uuid'" 4 in
+    try
+      start uuid_t;
+      let uuid = with_pbd P.get_uuid pbd in
+      if with_api P.get_by_uuid ~uuid = pbd then success uuid_t
+      else failed uuid_t "'get_by_uuid' failed";
+    with e -> debug uuid_t (ExnHelper.string_of_exn e); in
```
Comment (lindig) [inline ocaml/xapi/quicktest_bvt.ml]: I think this should fail the test.

File: ocaml/xapi/quicktest_bv_tests.ml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      let key,value = "key","value" in
+      with_obj add obj ~key ~value;
+
+      (* remove from other config *)
+      if List.mem (key,value) (with_obj gt obj)
+      then with_obj rm obj ~key (* may need ignore *)
+      else failed adrm_t "Error: add_to_other_config failed";
+
+      (* test fails if (key,value) has not been removed *)
+      if List.mem (key,value) (with_obj gt obj)
```
Comment (mseri) [inline ocaml/xapi/quicktest_bv_tests.ml]: Please add 2 keys and remove one, checking that the other is still present as well

File: ocaml/xapi/quicktest.ml [hunk 5]
```diff
@@ -877,6 +878,7 @@ let _ =
        (try
           maybe_run_test "encodings" Quicktest_encodings.run_from_within_quicktest;
           maybe_run_test "vm-memory-constraints" Quicktest_vm_memory_constraints.run_from_within_quicktest;
+          maybe_run_test "bvt" (fun () -> Quicktest_bv_tests.start s !rpc);
```
Comment (mseri) [inline ocaml/xapi/quicktest.ml]: Why are these called bvt?

Comment (minishrink) [reply ocaml/xapi/quicktest.ml]: The tests check low-level behaviour, so they're essentially basic verification tests. The plan is to add similar tests for calls affecting fields like device_config, generalising the test beyond just other_config calls.

**Pull request merged**